src/main/java/org/broadinstitute/hellbender/tools/exome/eval/EvaluateCopyNumberTriStateCalls.java

```diff
@@ -0,0 +1,772 @@
+            }
+        }
+        truthReader.close();
+        callsReader.close();
+        outputWriter.close();
+        closeCaseRecordWriter(caseWriter);
+        writeSampleSummaryFile(sampleSummaryOutputFile, sampleStats);
+
+        return "SUCCESS";
+    }
+
+    private void closeCaseRecordWriter(final GenotypeEvaluationRecordWriter caseWriter) {
+        try {
+           if (caseWriter != null) {
+               caseWriter.close();
+           }
+        } catch (final IOException ex) {
+            throw new UserException.CouldNotCreateOutputFile(caseDetailOutputFile, ex);
+        }
+    }
+
+    private void outputCases(final GenotypeEvaluationRecordWriter caseWriter, final VariantEvaluationContext vc, final TargetCollection<Target> targets) {
+        if (caseWriter == null) {
+            return;
+        }
+        for (final Genotype g : vc.getGenotypes()) {
+            final EvaluationClass evalClass =
+                    GATKProtectedVariantContextUtils.getAttributeAsObject(g, VariantEvaluationContext.EVALUATION_CLASS_KEY,
+                            EvaluationClass::parseString, null);
+            if (evalClass == null) {
+                continue;
+            }
+            final String sample = g.getSampleName();
+            final SimpleInterval interval = new SimpleInterval(vc);
+            final int targetCount = targets.targetCount(interval);
+            final Set<String> variantFilters = vc.getFilters();
+            final Genotype genotype = vc.getGenotype(sample);
+            final Set<String> genotypeFilterArray = genotype.getFilters() == null ? Collections.emptySet()
+                    : Stream.of(genotype.getFilters().split(VCFConstants.INFO_FIELD_ARRAY_SEPARATOR)).collect(Collectors.toSet());
+            final Set<String> allFilterStrings = new LinkedHashSet<>();
+            allFilterStrings.addAll(variantFilters);
+            allFilterStrings.addAll(genotypeFilterArray);
+            final Set<EvaluationFilter> allFilters = allFilterStrings.stream().map(EvaluationFilter::fromString).filter(Objects::nonNull).collect(Collectors.toSet());
+
+            final GenotypeEvaluationRecord record = new GenotypeEvaluationRecord(sample, interval, targetCount, evalClass, allFilters, vc);
+            try {
+                caseWriter.writeRecord(record);
+            } catch (final IOException ex) {
+                throw new UserException.CouldNotCreateOutputFile(this.caseDetailOutputFile, ex);
+            }
+        }
+    }
+
+    private void writeSampleSummaryFile(final File outputFile, final Map<String, EvaluationSampleSummaryRecord> sampleStats) {
+        if (sampleSummaryOutputFile == null) {
+            return;
+        }
+        try (final EvaluationSampleSummaryWriter writer =
+                     new EvaluationSampleSummaryWriter(outputFile, overallSampleSummaryRecordSampleName)) {
+            for (final EvaluationSampleSummaryRecord record : sampleStats.values()) {
```

Comment:
I would like this a bit more as a `forEach` one-liner.  Unless the `writer` has a `writeAll` method, which would be most preferable.